[4.0] Better css for frontend edit buttons

rdeutz opened the pull request:
Pull Request for Issue #32170 .

### Summary of Changes

A bit better css for the edit buttons,  I made the background half transparent and made the "Edit" text sr-only.

### Testing Instructions

* apply patch
*  node build/build.js --compile-css templates/cassiopeia/scss


### Actual result BEFORE applying this Pull Request
![grafik](https://user-images.githubusercontent.com/467356/111868006-d86bf780-8977-11eb-8c73-88e7183d6cdc.png)



### Expected result AFTER applying this Pull Request
#### LTR
![grafik](https://user-images.githubusercontent.com/467356/111904331-b2f9ef00-8a46-11eb-82fb-6215eb91d765.png)

#### RTL
![grafik](https://user-images.githubusercontent.com/467356/111904343-c73dec00-8a46-11eb-923b-a8f431fe1611.png)

### Documentation Changes Required
none

rdeutz committed:
better css for edit buttons

astridx commented:
Are you sure that `sr-only` is correct? 
I think in Boostrap 5 we should use `visually-hidden`
https://getbootstrap.com/docs/5.0/getting-started/accessibility/

drmenzelit reviewed (changes requested):
`<span class="vissually-hidden">`

brianteeman commented on layouts/joomla/edit/frontediting_modules.php:
```diff
@@ -40,7 +40,7 @@
 	// Create and add the edit link and tooltip
 	'\\1 <a class="btn btn-link jmodedit" href="' . $editUrl . '" target="' . $target . '" aria-describedby="tip-' . (int) $mod->id . '">
-	<span class="icon-edit" aria-hidden="true"></span>' . Text::_('JGLOBAL_EDIT') . '</a>
+	<span class="icon-edit" aria-hidden="true"></span><span class="sr-only">' . Text::_('JGLOBAL_EDIT') . '</span></a>
```
Please use visually-hidden instead

brianteeman commented on templates/cassiopeia/scss/blocks/_frontend-edit.scss:
```diff
@@ -2,15 +2,14 @@
   position: absolute;
-  right: 10px;
+  right: 0;
+  top:0;
```
```suggestion
  top: 0;
```

brianteeman commented on templates/cassiopeia/scss/blocks/_frontend-edit.scss:
```diff
@@ -2,15 +2,14 @@
   border-radius: .25rem;
 }
 
 a.jmenuedit {
```
might as well delete the class if it is empty

brianteeman commented:
It's a bit hard to tell from your screenshots but it looks like the issue with the placement of the breadcrumbs module edit link is not resolved. 

In screenshot 1 it is in the wrong place on the far right
In screenshot 2 I cant see it at all

brianteeman commented:
PS I had been thinking of a different approach but I havent had the time to go further with it and I doubt I will for some time.

Instead of the links/buttons being visible all the time we have a toggle button to display them. A bit like the accessibility plugin

infograf768 commented:
rtl

rdeutz committed:
fix for comments

rdeutz committed:
scss fix and bring button forward

rdeutz commented:
@infograf768 can you have a look at the rtl if my change does the trick, I don't have a site with rtl languages, seems they are all not ready for 4

rdeutz commented:
> It's a bit hard to tell from your screenshots but it looks like the issue with the placement of the breadcrumbs module edit link is not resolved.

It's not, it is in the top right corner. Don't seen atm how we can fix this with css.

brianteeman commented:
> @infograf768 can you have a look at the rtl if my change does the trick, I don't have a site with rtl languages, seems they are all not ready for 4

Persian is ready. Or just set change <rtl>0</rtl> in the active langmetadat.xml

rdeutz committed:
phpcs is a pain but scss lint is war

rdeutz committed:
revert changes

rdeutz committed:
position breadcrubs, fix rtl

rdeutz commented:
Ready to test, should now work.

brianteeman commented:
Nearly there.

Tested with the sample data (not the testing data)

- Breadcrumbs is now in the correct place
- Link in the top right - expected it to be the search module. It was actually the bottom menu module. Looks to be caused by the absolute positioning.
- The edit link for the search module is actually underneath (and therefore not selectable) the bottom menu module link. Again because of the absolute positioning

rdeutz committed:
hoping to have catched all

rdeutz commented:
> Link in the top right - expected it to be the search module. It was actually the bottom menu module. Looks to be caused by the absolute positioning.

fixed

> The edit link for the search module is actually underneath (and therefore not selectable) the bottom menu module link. Again because of the absolute positioning

fixed

brianteeman commented:
Lastest articles is still top right (and the tooltip is bottom right)

![image](https://user-images.githubusercontent.com/1296369/111907615-0b80ba80-8a4e-11eb-8597-9b84b7ff6a4e.png)

brianteeman commented:
mod_custom is also hidden in the top right

rdeutz committed:
hope dies last

rdeutz commented:
Next iteration to catch all modules.

rdeutz committed:
scss fix

chmst commented:
I have tested this item :white_check_mark: successfully on 62908f6b7cc19abcaa3f29063cdd44e13a23f5de<br /><br />Looks much better now. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32757">issues.joomla.org/tracker/joomla-cms/32757</a>.</sub>

drmenzelit commented:
I have tested this item :white_check_mark: successfully on 62908f6b7cc19abcaa3f29063cdd44e13a23f5de<br /><br />Tested with blog sample data.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32757">issues.joomla.org/tracker/joomla-cms/32757</a>.</sub>

rdeutz commented:
Set to RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32757">issues.joomla.org/tracker/joomla-cms/32757</a>.</sub>

drmenzelit commented:
Thanks
